### PR TITLE
SiPixel Payload Inspector: improve `SiPixelDynamicInefficiency` parametrization vs PU plotting

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

The goal of this PR is to introduce the possibility to display two curves of the pixel dynamic inefficiency parametrization vs inst. luminosity at the same time from the payload inspector service. Some graphical improvement for the display of the single payload viewer (introduced originally in https://github.com/cms-sw/cmssw/pull/40856) is also introduced.
The introduced class is also tested in the unit tests of this package.

#### PR validation:

Run:

```
getPayloadData.py \
    --plugin pluginSiPixelDynamicInefficiency_PayloadInspector \
    --plot plot_SiPixelDynamicInefficiencyPUParamComparisonTwoTags \
    --tag SiPixelDynamicInefficiency_phase1_2023_v2_fix3  \
    --tagtwo SiPixelDynamicInefficiency_phase1_2023_v2_fix2  \
    --time_type Run \
    --iovs '{"start_iov": "1", "end_iov": "1"}' \
    --iovstwo '{"start_iov": "1", "end_iov": "1"}' \
    --db Prod \
    --test;
```

and obtained the following plot:

image

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

N/A

### CODE REVIEW

**Comment by tvami** [comment]
@cmsbuild , please test

File: CondCore/SiPixelPlugins/plugins/SiPixelDynamicInefficiency_PayloadInspector.cc [hunk 1]
```diff
@@ -923,101 +1018,224 @@ namespace {
 
     static constexpr double xmin_ = 0.;   // x10e34 (min inst. lumi)
     static constexpr double xmax_ = 25.;  // x10e34 (max inst. lumi)
```
Comment (tvami) [inline CondCore/SiPixelPlugins/plugins/SiPixelDynamicInefficiency_PayloadInspector.cc:1020]: So this is not liked by CLANG

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by perrotta** [comment]
+1

**Pull request merged**